lib/openshift/service.rb

```diff
@@ -80,6 +80,21 @@ def ip(user: nil, cached: true, quiet: false)
       spec = raw_resource(user: user, cached: cached, quiet: quiet).dig('spec')
       return (spec.dig('portalIP') || spec.dig('clusterIP'))
     end
+    
+    # return service ipv6 address for dualstack cluster
+    def ip_v6(user: nil, cached: true, quiet: false)
+      spec = raw_resource(user: user, cached: cached, quiet: quiet).dig('spec')
+      ipv6 = spec['clusterIPs'].find { |ip| ip.include? ":" }
```

Comment:
And here as well about `[]` ? Or other option could be:
Check if the IP address does NOT contain `.` (dots) in that case it would be assumed to be ipv6

Would that be acceptable ?

Reply:
What do we think about ...

```ruby

    # return service ipv6 address as URL for dualstack cluster
    def ip_v6_url(user: nil, cached: true, quiet: false)
      return"[#{ip_v6(user: user, cached: cached, quiet: quiet)}]"
    end
```

Reply:
add one function ip_v6_url, see
```
    def ip_v6_url(user: nil, cached: true, quiet: false)
      ipv6 = self.ip_v6(user: user, cached: cached, quiet: quiet)
      port = self.ports(user: user, cached: true, quiet: quiet)[0]["port"]
      "[#{ipv6}]:#{port}"
    end
```